### PR TITLE
Add ability to have file extensions detected when present.

### PR DESCRIPTION
**Pull request by Bas-Man**
Provides support for three character support. Without breaking current tests.

### CODE REVIEW

**Comment by Bas-Man** [comment]
addresses #16

File: src/pattern.rs [hunk 1]
```diff
@@ -93,6 +93,7 @@ lazy_static! {
     pub static ref EPISODE: Pattern = regex!(
         r"(?i)e(?P<short>\d+)(?:[^\d]|$)|(episode)(?:[^\d]|$)(?P<long>\d+)|\d+x(?P<cross>\d+)|s\d+ - (?P<dash>\d+)"
     );
+    pub static ref FILE_EXTENSION: Pattern = regex!(r"(?i)(?:\.)(?P<extension>[a-z]{2,3}(?:\d)?)$");
```
Comment (IGI-111) [inline src/pattern.rs]: That's a cool trick, and it works surprisingly well from the samples I'm getting, but I'd like to also have support for `.h264`, `.webm`, `.m4v` and `.3gp`

Comment (Bas-Man) [reply src/pattern.rs]: I have never seen the first 2 as file extensions. But I can take a look and see if I can add them. I neglected `m4v`

If you have a list. I can see what I can do.

Comment (IGI-111) [reply src/pattern.rs]: Those should be enough. They're fairly rare as most people use Mpeg or Matroska containers but I've definitely seen some torrent with all of those.

Try to keep it in the same style though, new or esoteric video containers extension will probably always fall in that 2-4 alphanumeric category.

My main concern is avoiding collisions with group names which tend to also be short alphanumeric sequences at the end of bare names.

Comment (Bas-Man) [reply src/pattern.rs]: Something like this looks ok. It's not as clean as I would like

```
(?:\.)(?P<extension>[a-z]{2,4}(?:\d)?|m4v|3gp|h26[45])$
```

Note that this is only intended to match `files` 

such as 
```
something.avi
something.m4v
``` 

and will not match the existing examples and should NOT match. The `.` is part of the limiting condition.

Comment (Bas-Man) [reply src/pattern.rs]: Tested on [RustExp](https://rustexp.lpil.uk)
File extension reference [Wikipedia](https://en.wikipedia.org/wiki/Video_file_format)
scapshot

Let me know what you think.

**Comment by Bas-Man** [comment]
I broke the regex tests for extension to its own module so that it can be tested. I added a `m.extension() is None for all previous tests.

**Pull request merged**